src/bindings/js/node/src/async_infer_queue.cpp

```diff
@@ -0,0 +1,203 @@
+
+        for (size_t handle = 0; handle < jobs; handle++) {
+            m_requests.emplace_back(compiled.create_infer_request());
+            m_user_ids.push_back(std::make_pair(Napi::Reference<Napi::Object>::New(Napi::Object::New(env), 1),
+                                                Napi::Promise::Deferred::New(env)));
+            m_user_inputs.push_back(Napi::Reference<Napi::Object>::New(Napi::Object::New(env), 1));
+            m_idle_handles.push(handle);
+        }
+        this->set_default_callbacks();
+    } else {
+        reportError(info.Env(), "Invalid arguments. Expected CompiledModel and number of requests.");
+    }
+}
+
+AsyncInferQueue::~AsyncInferQueue() {
+    m_requests.clear();
+    m_user_ids.clear();
+    m_user_inputs.clear();
+}
+
+void AsyncInferQueue::release(const Napi::CallbackInfo& info) {
+    if (!this->tsfn) {
+        return;
+    }
+    const auto status = this->tsfn.Release();
+    if (status == napi_invalid_arg) {
+        reportError(info.Env(), "Failed to release AsyncInferQueue thread-safe function. Its thread-count is zero.");
+    } else if (status != napi_ok) {
+        reportError(info.Env(), "Failed to release AsyncInferQueue thread-safe function.");
+    }
+}
+
+Napi::Function AsyncInferQueue::get_class(Napi::Env env) {
+    return DefineClass(env,
+                       "AsyncInferQueue",
+                       {
+                           InstanceMethod("setCallback", &AsyncInferQueue::set_custom_callbacks),
+                           InstanceMethod("startAsync", &AsyncInferQueue::start_async),
+                           InstanceMethod("release", &AsyncInferQueue::release),
+                       });
+}
+
+int AsyncInferQueue::check_idle_request_id() {
+    std::unique_lock<std::mutex> lock(m_mutex);
+    if (m_idle_handles.empty()) {
+        return -1;
+    }
+    int idle_handle = static_cast<int>(m_idle_handles.front());
+    // wait for request to make sure it returned from callback
+    m_requests[idle_handle].wait();
+    m_idle_handles.pop();
+    return idle_handle;
+}
+
+void AsyncInferQueue::set_default_callbacks() {
+    for (size_t handle = 0; handle < m_requests.size(); handle++) {
+        m_requests[handle].set_callback([this, handle](std::exception_ptr exception_ptr) {
+            {
+                std::lock_guard<std::mutex> lock(m_mutex);
+                m_idle_handles.push(handle);
+            }
+            try {
+                if (exception_ptr) {
+                    std::rethrow_exception(exception_ptr);
+                }
+            } catch (const std::exception& e) {
+                OPENVINO_THROW(e.what());
+            }
+        });
+    }
+}
+
+void AsyncInferQueue::set_custom_callbacks(const Napi::CallbackInfo& info) {
+    std::vector<std::string> allowed_signatures;
+    try {
+        if (ov::js::validate<Napi::Function>(info, allowed_signatures)) {
+            this->tsfn = Napi::ThreadSafeFunction::New(info.Env(),
```

Comment:
```suggestion
            tsfn = Napi::ThreadSafeFunction::New(info.Env(),
```
Can `this` be removed where is not required?

Reply:
I removed all unrequired `this`